Alicerce/Sources/Logging/Log+NodeLogDestination.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  Log+NodeLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public class NodeLogDestination : LogDestination, LogDestinationFallible {
+
+        enum Error: Swift.Error {
+            case httpError(statusCode: Int)
+            case network(Swift.Error)
+        }
+
+        private static let dispatchQueueLabel = "com.mindera.alicerce.log.destination.node"
+        private static let defaultRequestTimeout: TimeInterval = 60
+
+        public let queue: Queue
+        public let minLevel: Level
+        public let formatter: LogItemFormatter
+        public private(set) var writtenItems: Int = 0
+
+        public var errorClosure: ((LogDestination, Item, Swift.Error) -> ())?
+
+        private let serverURL: URL
+        private let urlSession: URLSession
+        private let requestTimeout: TimeInterval
+
+        //MARK:- lifecycle
+
+        public init(serverURL: URL,
+                    minLevel: Level = .error,
+                    formatter: LogItemFormatter = StringLogItemFormatter(),
+                    urlSession: URLSession = URLSession.shared,
+                    queue: Queue = Queue(label: NodeLogDestination.dispatchQueueLabel),
+                    requestTimeout: TimeInterval = NodeLogDestination.defaultRequestTimeout) {
+
+            self.serverURL = serverURL
+            self.minLevel = minLevel
+            self.formatter = formatter
+            self.urlSession = urlSession
+            self.queue = queue
+            self.requestTimeout = requestTimeout
+        }
+
+        //MARK:- public methods
+
+        public func write(item: Item) {
+            queue.dispatchQueue.async { [weak self] in
+                guard let strongSelf = self else { return }
+
+                let formattedItem = strongSelf.formatter.format(logItem: item)
+                if let payloadData = formattedItem.data(using: .utf8) {
```

Comment:
I would be tempted to use a guard here, for two reasons:
1) We do nothing if `nil` is returned, so we may as well return early.
2) It would remove 1 level of indentation, which I find makes the code easier to read.

:smile: